scripts/tehn/playfair.lua

```diff
@@ -123,3 +141,17 @@ function redraw()
   end
   screen.update()
 end
+
+midi.add = function(dev)
+  print('playfair: midi device added', dev.id, dev.name)
+  midi_device = dev
+end
+midi.remove = function(dev)
+  print('playfair: midi device removed', dev.id, dev.name)
+  midi_device = nil
+end
+function cleanup()
+  midi_device = nil
+  midi.add = nil
+  midi.remove = nil
```

Comment:
you shouldn't need to null these either.  unless you found that you did? in which case that's a bug and should be fixed in norns proper (`script.lua` specifically).

/cc @artfwo @tehn

Reply:
without that cleanup function, i see print messages from the playfair handlers even after switching to other scripts. `playfair: midi device added` is printed when any other script loads.

Reply:
ok, wow.  thanks for confirming.

fwiw `gridkey` had a similar issue (https://github.com/monome/norns/commit/2e6762b409db0186156772137d1462734fc9ccd6) and it's an easy fix.  thanks for the catch!

Reply:
@pq hopefully the `register` thing will fix this for the future.

Reply:
in the meantime, here's what i think is a fix: https://github.com/monome/norns/pull/441